* grobian@gentoo.org: allow for compilation on Solaris, lazy approach to
                      resolve a conflict with FS from sys/regset.h

--- include/cdio++/iso9660.hpp
+++ include/cdio++/iso9660.hpp
@@ -35,6 +35,13 @@
 #include <cstdlib>
 using namespace std;
 
+/* On Solaris, FS is defined to an integer which makes compilation fail
+ * since we use that as identifier below.  It gets included via the
+ * chain sys/signal.h -> sys/ucontext.h -> sys/regset.h. */
+#ifdef FS
+# undef FS
+#endif
+
 /** ISO 9660 class.
 */
 class ISO9660
